prototype_source/pt2e_quantizer.rst

```diff
@@ -302,6 +302,106 @@ functions that are used in the example:
+      inputs, output = _find_input_and_output(match)
+      inputs[0].users[0].meta["quantization_annotation"] = ...
+      inputs[1].users[0].meta["quantization_annotation"] = ...
+      output.meta["quantization_annotation"] = ...
+
+With this, the ``Quantizer`` will still be valid even when the implementation for the ``nn`` modules and functionals changes, the ``aten`` IR for floating point model will change, but since we capture the pattern again instead of hardcoding the ``aten`` IR for the pattern, we'll get the updated ``aten`` IR as well and will still be able to match the pattern.
+
+One caveat is that if inputs of the pattern has multiple users, we don't have a good way to identify which user node we want to annotate except for checking the ``aten`` op target.
+
+Another caveat is that we need to make sure we have an exhaustive list of examples (such as 2D, 3D, 4D inputs, real vs. symbolic inputs, ``training=True`` vs. ``training=False``, and so on) for the pattern to make sure cover different possible ``aten`` IR outcomes captured from the ``torch`` IR pattern.
+      
+3. Using ``SubgraphMatcherWithNameNodeMap``
+----------------------------------------------
+We also introduced a different ``SubgraphMatcher`` utility called ``SubgraphMatcherWithNameNodeMap`` to make it easier to query the nodes that people want to annotate.
+
+Example::
+  
+  def conv_relu_pattern(x, weight, bias):
+      conv = torch.nn.functional.conv2d(x, weight, bias)
+      relu = torch.nn.functional.relu(conv)
+      # returns an additional dict that includes a map from name to node that we want to annotate
+      return relu, {"conv": conv, "relu": relu, "x": x, "weight": weight, "bias": bias}
+
+  matcher = SubgraphMatcherWithNameNodeMap(conv_relu_pattern)
+  matches = matcher.match(model)
+  for match in matches:
+      # find input and output of the pattern
+      # annotate the nodes
+      name_node_map = match.name_node_map
+      name_node_map["conv"].meta["quantization_annotation"] = ...
+      name_node_map["relu"].meta["quantization_annotation"] = ...
+
+This should be easier to use than the original ``SubgraphMatcher`` around finding the node we want to annotate or use. However, it may not work if some of the operators are captured as multiple ops. For example, if ``torch.nn.functional.conv2d`` is captured as multiple ops, then the inputs ``x``, ``weight``, ``bias`` may not be the direct inputs to the final operator marked by ``conv``, in this case we'll need to revert to the same way of access mentioned in the previous example.
+
+**Example:**::
+
+  for match in matches:
+      # find input and output of the pattern
+      # annotate the nodes
+      name_node_map = match.name_node_map
+      input_x = name_node_map["x"]
+      # since input_x may not be the input of name_node_map["conv"]
+      # we'll get the user nodes of this op and annotate that instead
+      input_x.users[0].meta["quantization_annotation"] = ...
+      ...
+
```

Comment:
Should we add (4) using `find_sequential_partitions`? I feel a lot of people will try to look at our `XNNPACKQuantizer` example when writing their own quantizers. It may be confusing to see a way to implement quantizers that are not described in the tutorials.

Alternatively, do we have plans to rewrite `XNNPACKQuantizer` to use either (2) or (3)?

Reply:
yeah I feel it's better to rewrite XNNPACKQuantizer to use (2) probably since (2) looks at graph instead of relying on metadata, cc @kimishpatel let me know what you think.

Reply:
I need to think if `find_sequential_partitions`, if its implementation is changed, will be useful to find some arbitrary sequential patterns. For now, it is not clear, so lets not add more confusion here.